### PR TITLE
Add parameter for expected HTTP code

### PR DESCRIPTION
**Pull request by dvorak**
Allow user to specify the expected HTTP result code.  Defaults to 200.

### CODE REVIEW

**Comment by EmilienM** [comment]
:+1:

File: lib/puppet/provider/http_conn_validator/http_conn_validator.rb [hunk 1]
```diff
@@ -58,7 +58,7 @@ def create
   #
   # @api private
   def validator
-    @validator ||= PuppetX::PuppetCommunity::HttpValidator.new(resource[:name], resource[:host], resource[:port], resource[:use_ssl], resource[:test_url])
+    @validator ||= PuppetX::PuppetCommunity::HttpValidator.new(resource[:name], resource[:host], resource[:port], resource[:use_ssl], resource[:test_url], resource[:expected_code])
```
Comment (igalic) [inline lib/puppet/provider/http_conn_validator/http_conn_validator.rb]: this is pretty awful :D

**Comment by igalic** [comment]
@dvorak would you mind also adding this new parameter to the readme?

**Comment by jyaworski** [comment]
Hello:

What's the status of this PR?

**Comment by EmilienM** [comment]
ping @dvorak

**Comment by dvorak** [comment]
Wrapped the ugly line I had made slightly uglier and added to README

**Comment by jyaworski** [comment]
LGTM.

**Pull request merged**